internal/storage/storage.go

```diff
@@ -0,0 +1,60 @@
+package storage
+
+import (
+	"time"
+)
+
+// Backend is an interface for defining functions which all storage providers support.
+type Backend interface {
+	Copy(file string) error
+	Prune(deadline time.Time, pruningPrefix string) (*PruneStats, error)
+	GetName() string
+}
+
+// StorageBackend is a generic type of storage. Everything here are common properties of all storage types.
+type StorageBackend struct {
+	Backend
```

Comment:
I don't think it's needed to embed the interface type here, is it?

Reply:
It was only needed for the DoPrune function so it is able to access the Name() func. Changed it use a function parameter now, which eliminates the need for the interface inside the struct.